modules/runner/src/main/ts/core/Reporter.ts

```diff
@@ -0,0 +1,163 @@
+import { LoggedError, Reporter } from '@ephox/bedrock-common';
+import { Callbacks, onSuccessCallback } from './Callbacks';
+import { UrlParams } from './UrlParams';
+import { formatElapsedTime, makeUrl } from './Utils';
+
+export interface TestReporter {
+  start: (onDone: onSuccessCallback) => void;
+  pass: (onDone: onSuccessCallback) => void;
+  fail: (e: LoggedError.LoggedError, onDone: onSuccessCallback) => void;
+}
+
+export interface Reporter {
+  summary: () => { passed: number; failed: number };
+  test: (file: string, name: string) => TestReporter;
+  done: () => void;
+  setStopOnFailure: (stopOnFailure: boolean) => void;
+  shouldStopOnFailure: () => boolean;
+}
+
+declare const $: JQueryStatic;
+
+const elapsed = (since: Date): string => formatElapsedTime(since, new Date());
+
+export const dom = (container: JQuery<HTMLElement>, totalNumTests: number, params: UrlParams, callbacks: Callbacks): Reporter => {
+  const initial = new Date();
+  let passCount = 0;
+  let failCount = 0;
+  let stopOnFailure = false;
+
+  const summary = () => ({
+    passed: passCount + (params.offset - params.failed),
+    failed: failCount + params.failed,
+  });
+
+  const current = $('<span />').addClass('progress').text(params.offset);
+  const restartBtn = $('<button />').text('Restart').on('click', () => {
+    const url = makeUrl('', 0, 0, 0);
+    window.location.assign(url);
```

Comment:
Minor: you could separate out the DOM entirely and pass in the handlers for the commands, and have setters for the labels.

Reply:
Yeah, I really should have continued to split this out more. I got so far and was running out of time I allotted to work on this to be honest. I'll have another go and see how far I can get :+1:

Reply:
I'll use `Really, really, really, really, really minor` next time so you feel less pressure to make changes to something that was already not in your sprint :)

Reply:
Okay, have done a fair bit more reworking. How does that look now?